src/sql/workbench/contrib/query/test/browser/queryInputFactory.test.ts

```diff
@@ -112,7 +195,90 @@ suite('Query Input Factory', () => {
 		assert(isThenable(response));
 		await response;
 		assert(newsqlEditorStub.calledWithExactly({ resource: undefined, open: false, initalContent: '' }));
-		assert(connectionManagementService.numberConnects === 1, 'Convert input should have called connect when active editor connection exists');
+		assert(connectionManagementService.numberConnects === 1, 'Async convert input should have called connect only once for one URI');
+	});
+
+	test('only one sync query editor input can call connect with a unique uri when global connection exists (Editor)', () => {
+		instantiationService = workbenchInstantiationService();
+		const editorService = new MockEditorService(instantiationService);
+		const connectionManagementService = new MockConnectionManagementService();
+		instantiationService.stub(IObjectExplorerService, new MockObjectExplorerService());
+		instantiationService.stub(IConnectionManagementService, connectionManagementService);
+		instantiationService.stub(IEditorService, editorService);
+		const queryEditorLanguageAssociation = instantiationService.createInstance(QueryEditorLanguageAssociation);
+		const input1 = createFileInput(URI.file('/test/file.sql'), undefined, undefined, undefined);
+		const input2 = createFileInput(URI.file('/test/file.sql'), undefined, undefined, undefined);
+		queryEditorLanguageAssociation.syncConvertInput(input1);
+		queryEditorLanguageAssociation.syncConvertInput(input2);
+		let connProfile = connectionManagementService.getConnectionProfile('file:///test/file.sql');
+		assert(connProfile !== undefined, 'connection profile should not be undefined');
+		assert(connectionManagementService.numberConnects === 1, 'Sync convert input should have called connect only once for one URI');
+	});
+
+	test('sync query editor inputs can be connected one after the other with different uris when global connection exists (Editor)', () => {
+		instantiationService = workbenchInstantiationService();
+		const editorService = new MockEditorService(instantiationService);
+		const connectionManagementService = new MockConnectionManagementService();
+		instantiationService.stub(IObjectExplorerService, new MockObjectExplorerService());
+		instantiationService.stub(IConnectionManagementService, connectionManagementService);
+		instantiationService.stub(IEditorService, editorService);
+		const queryEditorLanguageAssociation = instantiationService.createInstance(QueryEditorLanguageAssociation);
+		const input1 = createFileInput(URI.file('/test/file1.sql'), undefined, undefined, undefined);
+		const input2 = createFileInput(URI.file('/test/file2.sql'), undefined, undefined, undefined);
+		queryEditorLanguageAssociation.syncConvertInput(input1);
+		queryEditorLanguageAssociation.syncConvertInput(input2);
+		let connProfile1 = connectionManagementService.getConnectionProfile('file:///test/file1.sql');
```

Comment:
have you run these tests on both windows and mac at least? the path resolution might have some issues here

Reply:
Tests were run on Windows just fine, Mac is fine according to CI, I am going to investigate on my Mac itself if that is needed

Reply:
Works fine on my actual Mac, so far, seems good.